.azure/templates/build-test.yml

```diff
@@ -118,6 +118,7 @@ steps:
       cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
             -DCMAKE_INSTALL_PREFIX=install \
             -DCMAKE_PREFIX_PATH="${BUILD_SOURCESDIRECTORY}/cyclonedds/build/install;${BUILD_SOURCESDIRECTORY}/iceoryx/build/install;${BUILD_SOURCESDIRECTORY}/googletest/build/install" \
+            -DCYCLONE_SOURCE_DIR="${BUILD_SOURCESDIRECTORY}/cyclonedds" \
```

Comment:
I suspect this is needed for including `dds__sysdef_model.h` in the test code, but it is really inelegant to require the C sources to be present to build the C++ binding — we never had that tight a relationship, and I don't think the QoS provider should change that.

Now, I haven't made an effort yet to find out which bits exactly are needed but it appears likely that only a fraction of what is in that file is needed. Before merging it like this, I'd like to know whether the bits that are needed can reasonably be installed by the core library. That would eliminate the need for this hack.

(Apologies for not spotting this earlier.)